service.go

```diff
@@ -47,6 +47,8 @@ type Service interface {
 	// should be used. The returned error will be formatted as a websocket
 	// error code 4000, using the string form of the error as the message.
 	ProcessClientRequest(req *http.Request, handler string, msg []byte) (reply []byte, tunnel *StreamingTunnel, err error)
+	ProcessClientRequest2(req *http.Request, path string, buf []byte, furtherInputs chan []byte) ([]byte, *StreamingTunnel, error)
+	IsStreaming(path string) (bool, error)
```

Comment:
Adding these two methods to the `Service` interface breaks all the existing services!

Another way to do it is to define a `BidirectionalStreamer interface` with only the `ProcessClientRequest2` in it (with a better name). Then, instead of doing `if s.IsStreaming` you can do 

```
if bisi, ok := s.(*BidirectionalStreamer){
  bisi.ProcessClientRequest2....
} else {
  s.ProcessClientRequest
}
```